src/v/debug_bundle/debug_bundle_service.cc

```diff
@@ -755,5 +792,81 @@ ss::future<> service::handle_wait_result(job_id_t job_id) {
         co_return;
     }
 }
+ss::future<> service::maybe_reload_previous_run() {
+    auto md_buf = _kvstore->get(
+      storage::kvstore::key_space::debug_bundle,
+      bytes::from_string(debug_bundle_metadata_key));
+
+    if (!md_buf) {
+        vlog(lg.debug, "No previous run detected");
+        co_return;
+    }
+
+    iobuf_parser p(std::move(*md_buf));
+    auto md = serde::read<metadata>(p);
+
+    auto run_was_successful = [&md]() {
+        auto wait_status = md.get_wait_status();
+        return std::holds_alternative<ss::experimental::process::wait_exited>(
+                 wait_status)
+               && std::get<ss::experimental::process::wait_exited>(wait_status)
+                      .exit_code
+                    == 0;
+    }();
+
+    if (
+      run_was_successful
+      && !co_await ss::file_exists(md.debug_bundle_file_path)) {
+        vlog(
+          lg.debug,
+          "Debug bundle file {} does not exist, cannot reload metadata",
+          md.debug_bundle_file_path);
+        co_return co_await remove_kvstore_entry();
+    }
+
+    if (
+      run_was_successful
+      && !co_await validate_sha256_checksum(
+        md.debug_bundle_file_path, md.sha256_checksum)) {
+        vlog(
+          lg.debug,
+          "Debug bundle file {} checksum mismatch",
+          md.debug_bundle_file_path);
+        co_await ss::remove_file(md.debug_bundle_file_path);
+        co_return co_await remove_kvstore_entry();
+    }
+
+    vlog(
+      lg.info,
+      "Detected a valid previous run that was {}successful",
+      run_was_successful ? "" : "un");
+
+    std::optional<process_output> po;
+    if (co_await ss::file_exists(md.process_output_file_path)) {
+        try {
+            auto buf = co_await read_file(md.process_output_file_path);
+            iobuf_parser p(std::move(buf));
+            po.emplace(serde::read<process_output>(p));
+        } catch (const std::exception& e) {
+            vlog(
+              lg.warn,
+              "Failed to read process output from {}: {}",
+              md.process_output_file_path,
+              e.what());
+        }
+    }
+
+    if (!po) {
+        vlog(
```

Comment:
Is it worth doing cleanup?

Reply:
good catch here, needs some better tidying